server/models/Integration.ts

```diff
@@ -54,17 +38,17 @@ export type IntegrationSettings =
 }))
 @Table({ tableName: "integrations", modelName: "integration" })
 @Fix
-class Integration extends IdModel {
+class Integration<T> extends IdModel {
   @IsIn([Object.values(IntegrationType)])
-  @Column
-  type: string;
+  @Column(DataType.STRING)
+  type: T;
```

Comment:
Keeping it `T` so the type inference of settings below is dependent upon T(during usage, `T` is supposed to be passed as one of `IntegrationType.Embed`, `IntegrationType.Post` and `IntegrationType.Command`).

Reply:
Reverted due to type recurring type incompatibility issues.